src/Internal/Payment/Router.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Class Router
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment;
+
+use WCPay\Core\Server\Request\Get_Payment_Process_Factors;
+use WCPay\Database_Cache;
+use WCPay\Exceptions\API_Exception;
+use WCPay\Internal\Payment\Factor;
+
+/**
+ * Until the new payment process is fully developed, and the legacy
+ * process is gone, the new process is managed as a feature behind a router.
+ *
+ * This class will be removed once the new process is the default
+ * option for all payments.
+ */
+class Router {
+	/**
+	 * Database cache.
+	 *
+	 * @var Database_Cache
+	 */
+	protected $database_cache;
+
+	/**
+	 * Class constructor, receiving dependencies.
+	 *
+	 * @param Database_Cache $database_cache Database cache.
+	 */
+	public function __construct( Database_Cache $database_cache ) {
+		$this->database_cache = $database_cache;
+	}
+
+	/**
+	 * Checks whether a given payment should use the new payment process.
+	 *
+	 * @param Factor[] $factors Factors, describing the type and conditions of the payment.
+	 * @return bool
+	 * @psalm-suppress MissingThrowsDocblock
+	 */
+	public function should_use_new_payment_process( array $factors ): bool {
+		$allowed_factors = $this->get_allowed_factors();
+
+		// This would make sure that the payment process is a factor as well.
+		// phpcs:ignore WordPress.PHP.StrictInArray.FoundNonStrictFalse
+		if ( ! in_array( Factor::NEW_PAYMENT_PROCESS(), $factors, false ) ) {
+			$factors[] = Factor::NEW_PAYMENT_PROCESS();
+		}
```

Comment:
> this will be called in only one place and removed altogether in the future.

Actually, this routing will be used in a few places before being removed:

- The post-3DS-authentication part of the flow
- Subscription renewal
- I'd like IPP payments to also start sharing the payment process. [Some methods](https://github.com/Automattic/woocommerce-payments/blob/rpp/6679-factor-flags/includes/admin/class-wc-rest-payments-orders-controller.php#L138) are very similar to `process_payment`, and have basically copy-pasted code right now.
- Similarly, Link payments could also be linked there.

With that clarified, I realize that I'm listing specific flows, which have their own factor flag. Their entry points will always have `Factor::SOME_FLAG() => true` hard-coded. I'm removing this condition...

Reply:
Done in https://github.com/Automattic/woocommerce-payments/pull/7035/commits/75bfaa28805cee399aea2e1405aa53b881c708a9